### ISSUE TITLE
Security feature: Show seed phrase only after entering password

### ISSUE THREAD

**Issue by oPFGKk9gtuw8nuHkzrQn** [joinmarket-webui/jam#872]
### **Is your feature request related to a problem? Please describe.**

When logged into JAM, any user of that computer can easily access the seed phrase by simply selecting "Show seed phrase" as shown in the attached screenshot. This presents a significant security risk, as there is no additional authentication step required to view this sensitive information. This lack of a security barrier could lead to unauthorized access to the wallet if someone else gains temporary access to the browser window.
![grafik](https://github.com/user-attachments/assets/5c1e092b-aa0d-458d-8836-b76b8e140316)
![grafik](https://github.com/user-attachments/assets/940377fa-9c9f-4383-b972-283cbf3b6063)

### **Describe the solution you'd like**

Implement an additional layer of security for displaying the seed phrase. Specifically, when a user attempts to view the seed phrase, they should be prompted to enter their wallet password again. This step would ensure that only the legitimate wallet owner can view the seed phrase, adding an essential security checkpoint.

### **Describe alternatives you've considered**

 **Timed Access:** Allow the seed phrase to be visible for only a short period (e.g., 30 seconds) after successful authentication, after which the user would need to re-authenticate to view it again. Or lock the browser window after a certain amount of time.

### **Additional context**

- **Screenshot**: Attached screenshot shows the current "Show seed phrase" option without any security prompt.
- **User Impact**: Enhancing this feature would significantly improve the security posture of JAM, protecting users from potential theft or unauthorized access to their cryptocurrency wallets.

**Comment by MarnixCroes** [comment]
well, if someone has access to your device they could also send all the money out.

**Comment by khadar1020** [comment]
I would like to work on this @theborakompanioni can you assign me this issue

**Comment by khadar1020** [comment]
https://github.com/user-attachments/assets/dd5b5db8-dfb2-47ed-932f-5a9c736ad667

This is the video @theborakompanioni I have solved this issue if any changes you want I will be doing other changes please tell me other wise I am ready to submit a PR 

Thank you

**Comment by parrth20** [comment]
I would be interested in working on this. Could you please assign it to me?

**Comment by theborakompanioni** [comment]
Hey @parrth20! This is also already worked on and has two PRs open: https://github.com/joinmarket-webui/jam/pull/874 and https://github.com/joinmarket-webui/jam/pull/881.

It seems https://github.com/joinmarket-webui/jam/pull/881is of higher quality and #874 should be closed.
[Maybe you can look at the review question](https://github.com/joinmarket-webui/jam/pull/881#issuecomment-2740860201) and investigate whether this is a problem? 💪

**Comment by rushil23451** [comment]
Hey @theborakompanioni i have looked at the question you asked on the thread and have already started to look at it, will get back to you with a relevant solution shortly, was busy with some work

**Comment by parrth20** [comment]
@rushil23451 @theborakompanioni , your PR is almost correct; however, you don't need to use that token again, as it could create problems in the long run.

**Comment by theborakompanioni** [comment]
> [@rushil23451](https://github.com/rushil23451) [@theborakompanioni](https://github.com/theborakompanioni) , your PR is almost correct; however, you don't need to use that token again, as it could create problems in the long run.

Can you clarify what you mean by that? What can create problem? Using the token? From my point of view it looks like it is a problem **not** using it.

**Comment by rushil23451** [comment]
@theborakompanioni  Yes the session refreshes every 15-30minutes(need to check it again) we need to store the refresh tokens properly so that the user doesn’t get locked out of the system, will implement the change and store the refresh tokens properly

**Comment by theborakompanioni** [comment]
> [@theborakompanioni](https://github.com/theborakompanioni) Yes the session refreshes every 15-30minutes(need to check it again) we need to store the refresh tokens properly so that the user doesn’t get locked out of the system, will implement the change and store the refresh tokens properly

Yeah. Please check that 🙏 I don't even know if the current method is working properly in all cases. Some users reported they have been mysteriously logged out. Low severity as any background job still runs, but they have to "log in" again, which is annoying.

**Comment by parrth20** [comment]
Hey Sir @theborakompanioni ,
This is how i want to approach i need few of this to be clarified to please lmk that 
 
I’d like to implement this and open a PR.


Confirm scope: “Password re-entry required every time the seed phrase is revealed (even if wallet is already unlocked).”
Ask preference: “Should the re-auth be (a) every reveal, (b) once per session with timeout, or (c) once per page-load?”
plan: “I’ll add a password prompt modal, call an API to validate/retrieve mnemonic, show phrase only after success, and clear it on close.”

**Comment by parrth20** [comment]
> Hey Sir [@theborakompanioni](https://github.com/theborakompanioni) , This is how i want to approach i need few of this to be clarified to please lmk that
> 
> I’d like to implement this and open a PR.
> 
> Confirm scope: “Password re-entry required every time the seed phrase is revealed (even if wallet is already unlocked).” Ask preference: “Should the re-auth be (a) every reveal, (b) once per session with timeout, or (c) once per page-load?” plan: “I’ll add a password prompt modal, call an API to validate/retrieve mnemonic, show phrase only after success, and clear it on close.”

only if this issue is still open,

**Comment by theborakompanioni** [comment]
> > Hey Sir [@theborakompanioni](https://github.com/theborakompanioni) , This is how i want to approach i need few of this to be clarified to please lmk that
> > I’d like to implement this and open a PR.
> > Confirm scope: “Password re-entry required every time the seed phrase is revealed (even if wallet is already unlocked).” Ask preference: “Should the re-auth be (a) every reveal, (b) once per session with timeout, or (c) once per page-load?” plan: “I’ll add a password prompt modal, call an API to validate/retrieve mnemonic, show phrase only after success, and clear it on close.”
> 
> only if this issue is still open,

Hey @parrth20! This is already implemented in the v2 branch. Closing the issue, thanks for pointing it out 🙏
